build-system/tasks/serve.js

```diff
@@ -150,7 +148,7 @@ async function serve() {
   logServeMode();
   watch(serverFiles, restartServer);
   await startServer();
```

Comment:
Could you `await Promise.all([startServer(), performPreBuildSteps()])`?

Reply:
Tried that initially. Server startup takes ~1 second, so I separated the steps in order to prevent the logs from getting interspersed.